includes/support/cartbounty.php

```diff
@@ -0,0 +1,310 @@
+        'cartbounty_shipping_state'      => $razorpayData['customer_details']['shipping_address']['state'] ?? '',
+        'cartbounty_shipping_postcode'   => $razorpayData['customer_details']['shipping_address']['zipcode'] ?? '',
+        'cartbounty_order_comments'      => '',
+        'cartbounty_create_account'      => '',
+        'cartbounty_ship_elsewhere'      => '',
+    );
+
+    $userData = array(
+        'name'         => $name,
+        'surname'      => $surname,
+        'email'        => $email,
+        'phone'        => $phone,
+        'location'     => $location,
+        'other_fields' => $otherFields,
+    );
+
+    $sessionID = getSessionID($razorpayData['receipt']);
+
+    if ($cartbountyPublic->cart_saved($sessionID) == false) {
+        //If cart is not saved
+        $wpdb->query(
+            $wpdb->prepare(
+                "INSERT INTO $cartTable
+          ( name, surname, email, phone, location, cart_contents, cart_total, currency, time, session_id, other_fields)
+          VALUES ( %s, %s, %s, %s, %s, %s, %0.2f, %s, %s, %s, %s)",
+                array(
+                    'name'         => sanitize_text_field($userData['name']),
+                    'surname'      => sanitize_text_field($userData['surname']),
+                    'email'        => sanitize_email($userData['email']),
+                    'phone'        => filter_var($userData['phone'], FILTER_SANITIZE_NUMBER_INT),
+                    'location'     => sanitize_text_field(serialize($userData['location'])),
+                    'products'     => serialize($cart['product_array']),
+                    'total'        => sanitize_text_field($cart['cart_total']),
+                    'currency'     => sanitize_text_field($cart['cart_currency']),
+                    'time'         => sanitize_text_field($cart['current_time']),
+                    'session_id'   => sanitize_text_field($cart['session_id']),
+                    'other_fields' => sanitize_text_field(serialize($userData['other_fields'])),
+                )
+            )
+        );
+
+        $cartbountyPublic->increase_recoverable_cart_count();
+        $cartbountyPublic->set_cartbounty_session($sessionID);
+    }
+    $updatedRows = $wpdb->query(
+        $wpdb->prepare(
+            "UPDATE $cartTable
+        SET name     = %s,
+        surname      = %s,
+        email        = %s,
+        phone        = %s,
+        location     = %s,
+        other_fields = '$otherFields'
+        WHERE session_id = %s and type=0",
+            sanitize_text_field($userData['name']),
+            sanitize_text_field($userData['surname']),
+            sanitize_email($userData['email']),
+            filter_var($userData['phone'], FILTER_SANITIZE_NUMBER_INT),
+            sanitize_text_field(serialize($userData['location'])),
+            $sessionID
+        )
+    );
+
+    deleteDuplicateCarts($sessionID, $updatedRows, $cartTable);
```

Comment:
Still not using plugin function?

Reply:
We can't use it as it is throwing critical errors.
This is a private function within it and the rest are public ones and we're able to reuse them.